waiter/src/waiter/marathon.clj

```diff
@@ -75,6 +76,29 @@
        (finally
          (.close sw#)))))
 
+(defn- get-deployment-info
+  "Extracts the deployments section from the response body if it exists."
+  [{:keys [body]}]
+  (-> body
+      str
+      json/read-str
+      (get "deployments")
+      not-empty))
+
+(defn extract-deployment-info
+  "If the response body has deployments listed, pings Marathon to get the deployment details."
+  [response]
+  (try
+    (when-let [deployments (get-deployment-info response)]
+      (let [deployment-ids (->> deployments
+                                (map #(get % "id"))
+                                set)
+            all-marathon-deployments (md/get-deployments)]
+        (->> all-marathon-deployments
+             (filter (fn [{:strs [id]}] (contains? deployment-ids id)))
+             vec)))
+    (catch Exception _ (comment "gobble the exception"))))
```

Comment:
Why not log it so we can know what the data was if parsing fails or something else happens?

Reply:
Done.